Not compiling, again )

Hi all,
pulled latest v8 from github, installed, copied all headers and libs, make fails with the following output:

``````
/bin/bash /tmp/v8js/libtool --mode=compile g++ -std=c++11 -I. -I/tmp/v8js -DPHP_ATOM_INC -I/tmp/v8js/include -I/tmp/v8js/main -I/tmp/v8js -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr/include/php5/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /tmp/v8js/v8js.cc -o v8js.lo 
libtool: compile:  g++ -std=c++11 -I. -I/tmp/v8js -DPHP_ATOM_INC -I/tmp/v8js/include -I/tmp/v8js/main -I/tmp/v8js -I/usr/include/php5 -I/usr/include/php5/main -I/usr/include/php5/TSRM -I/usr/include/php5/Zend -I/usr/include/php5/ext -I/usr/include/php5/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /tmp/v8js/v8js.cc  -fPIC -DPIC -o .libs/v8js.o
/tmp/v8js/v8js.cc:80:1: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
/tmp/v8js/v8js.cc:80:1: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
/tmp/v8js/v8js.cc:80:1: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h: In function 'zval* php_v8js_v8_read_property(zval*, zval*, int, const zend_literal*)':
/usr/include/v8.h:771:3: error: 'T* v8::Persistent<T>::operator->() const [with T = v8::Value]' is private
/tmp/v8js/v8js.cc:134:49: error: within this context
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:771:3: error: 'T* v8::Persistent<T>::operator->() const [with T = v8::Value]' is private
/tmp/v8js/v8js.cc:134:76: error: within this context
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:771:3: error: 'T* v8::Persistent<T>::operator->() const [with T = v8::Value]' is private
/tmp/v8js/v8js.cc:136:43: error: within this context
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h: In function 'void php_v8js_v8_write_property(zval*, zval*, zval*, const zend_literal*)':
/usr/include/v8.h:771:3: error: 'T* v8::Persistent<T>::operator->() const [with T = v8::Value]' is private
/tmp/v8js/v8js.cc:167:16: error: within this context
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:771:3: error: 'T* v8::Persistent<T>::operator->() const [with T = v8::Value]' is private
/tmp/v8js/v8js.cc:167:43: error: within this context
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:771:3: error: 'T* v8::Persistent<T>::operator->() const [with T = v8::Value]' is private
/tmp/v8js/v8js.cc:168:13: error: within this context
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h: In function 'void php_v8js_v8_unset_property(zval*, zval*, const zend_literal*)':
/usr/include/v8.h:771:3: error: 'T* v8::Persistent<T>::operator->() const [with T = v8::Value]' is private
/tmp/v8js/v8js.cc:177:16: error: within this context
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:771:3: error: 'T* v8::Persistent<T>::operator->() const [with T = v8::Value]' is private
/tmp/v8js/v8js.cc:177:43: error: within this context
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:771:3: error: 'T* v8::Persistent<T>::operator->() const [with T = v8::Value]' is private
/tmp/v8js/v8js.cc:178:13: error: within this context
/tmp/v8js/v8js.cc: In function 'HashTable* php_v8js_v8_get_properties(zval*)':
/tmp/v8js/v8js.cc:234:62: warning: 'static v8::Persistent<v8::Context> v8::Context::New(v8::ExtensionConfiguration*, v8::Handle<v8::ObjectTemplate>, v8::Handle<v8::Value>)' is deprecated (declared at /usr/include/v8.h:4925) [-Wdeprecated-declarations]
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:751:3: error: 'v8::Persistent<T>::Persistent(const v8::Persistent<T>&) [with T = v8::Context; v8::Persistent<T> = v8::Persistent<v8::Context>]' is private
/tmp/v8js/v8js.cc:234:62: error: within this context
/tmp/v8js/v8js.cc:235:44: error: no matching function for call to 'v8::Context::Scope::Scope(v8::Persistent<v8::Context>&)'
/tmp/v8js/v8js.cc:235:44: note: candidates are:
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:5046:5: note: v8::Context::Scope::Scope(v8::Isolate*, v8::Persistent<v8::Context>&)
/usr/include/v8.h:5046:5: note:   candidate expects 2 arguments, 1 provided
/usr/include/v8.h:5042:14: note: v8::Context::Scope::Scope(v8::Handle<v8::Context>)
/usr/include/v8.h:5042:14: note:   no known conversion for argument 1 from 'v8::Persistent<v8::Context>' to 'v8::Handle<v8::Context>'
/usr/include/v8.h:5040:9: note: constexpr v8::Context::Scope::Scope(const v8::Context::Scope&)
/usr/include/v8.h:5040:9: note:   no known conversion for argument 1 from 'v8::Persistent<v8::Context>' to 'const v8::Context::Scope&'
/tmp/v8js/v8js.cc:237:92: error: could not convert 'obj->php_v8js_object::v8obj' from 'v8::Persistent<v8::Value>' to 'v8::Handle<v8::Value>'
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h: In function 'zend_function* php_v8js_v8_get_method(zval**, char*, int, const zend_literal*)':
/usr/include/v8.h:771:3: error: 'T* v8::Persistent<T>::operator->() const [with T = v8::Value]' is private
/tmp/v8js/v8js.cc:260:41: error: within this context
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:771:3: error: 'T* v8::Persistent<T>::operator->() const [with T = v8::Value]' is private
/tmp/v8js/v8js.cc:260:68: error: within this context
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:771:3: error: 'T* v8::Persistent<T>::operator->() const [with T = v8::Value]' is private
/tmp/v8js/v8js.cc:261:43: error: within this context
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h: In function 'int php_v8js_v8_call_method(const char*, int, zval*, zval**, zval*, int)':
/usr/include/v8.h:771:3: error: 'T* v8::Persistent<T>::operator->() const [with T = v8::Value]' is private
/tmp/v8js/v8js.cc:298:42: error: within this context
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h: In function 'int php_v8js_v8_get_closure(zval*, zend_class_entry**, zend_function**, zval**)':
/usr/include/v8.h:771:3: error: 'T* v8::Persistent<T>::operator->() const [with T = v8::Value]' is private
/tmp/v8js/v8js.cc:336:17: error: within this context
/tmp/v8js/v8js.cc: In function 'void php_v8js_create_v8(zval*, v8::Handle<v8::Value>, int, v8::Isolate*)':
/tmp/v8js/v8js.cc:394:49: error: no matching function for call to 'v8::Persistent<v8::Value>::New(v8::Handle<v8::Value>&)'
/tmp/v8js/v8js.cc:394:49: note: candidate is:
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:5585:4: note: static T* v8::Persistent<T>::New(v8::Isolate*, T*) [with T = v8::Value]
/usr/include/v8.h:5585:4: note:   candidate expects 2 arguments, 1 provided
/tmp/v8js/v8js.cc: In function 'void zim_V8Js___construct(int, zval*, zval**, zval*, int)':
/tmp/v8js/v8js.cc:592:92: error: no matching function for call to 'v8::Persistent<v8::FunctionTemplate>::New(v8::Local<v8::FunctionTemplate>)'
/tmp/v8js/v8js.cc:592:92: note: candidate is:
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:5585:4: note: static T* v8::Persistent<T>::New(v8::Isolate*, T*) [with T = v8::FunctionTemplate]
/usr/include/v8.h:5585:4: note:   candidate expects 2 arguments, 1 provided
/usr/include/v8.h:771:3: error: 'T* v8::Persistent<T>::operator->() const [with T = v8::FunctionTemplate]' is private
/tmp/v8js/v8js.cc:593:20: error: within this context
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:771:3: error: 'T* v8::Persistent<T>::operator->() const [with T = v8::FunctionTemplate]' is private
/tmp/v8js/v8js.cc:596:46: error: within this context
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:771:3: error: 'T* v8::Persistent<T>::operator->() const [with T = v8::FunctionTemplate]' is private
/tmp/v8js/v8js.cc:599:67: error: within this context
/tmp/v8js/v8js.cc:599:87: warning: 'static v8::Persistent<v8::Context> v8::Context::New(v8::ExtensionConfiguration*, v8::Handle<v8::ObjectTemplate>, v8::Handle<v8::Value>)' is deprecated (declared at /usr/include/v8.h:4925) [-Wdeprecated-declarations]
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:753:3: error: 'v8::Persistent<T>& v8::Persistent<T>::operator=(const v8::Persistent<T>&) [with T = v8::Context; v8::Persistent<T> = v8::Persistent<v8::Context>]' is private
/tmp/v8js/v8js.cc:599:87: error: within this context
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:771:3: error: 'T* v8::Persistent<T>::operator->() const [with T = v8::Context]' is private
/tmp/v8js/v8js.cc:600:12: error: within this context
/tmp/v8js/v8js.cc:614:45: error: no matching function for call to 'v8::Context::Scope::Scope(v8::Persistent<v8::Context>&)'
/tmp/v8js/v8js.cc:614:45: note: candidates are:
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:5046:5: note: v8::Context::Scope::Scope(v8::Isolate*, v8::Persistent<v8::Context>&)
/usr/include/v8.h:5046:5: note:   candidate expects 2 arguments, 1 provided
/usr/include/v8.h:5042:14: note: v8::Context::Scope::Scope(v8::Handle<v8::Context>)
/usr/include/v8.h:5042:14: note:   no known conversion for argument 1 from 'v8::Persistent<v8::Context>' to 'v8::Handle<v8::Context>'
/usr/include/v8.h:5040:9: note: constexpr v8::Context::Scope::Scope(const v8::Context::Scope&)
/usr/include/v8.h:5040:9: note:   no known conversion for argument 1 from 'v8::Persistent<v8::Context>' to 'const v8::Context::Scope&'
/tmp/v8js/v8js.cc:637:128: error: no matching function for call to 'v8::Persistent<v8::String>::New(v8::Local<v8::String>)'
/tmp/v8js/v8js.cc:637:128: note: candidate is:
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:5585:4: note: static T* v8::Persistent<T>::New(v8::Isolate*, T*) [with T = v8::String]
/usr/include/v8.h:5585:4: note:   candidate expects 2 arguments, 1 provided
/tmp/v8js/v8js.cc:640:93: error: no matching function for call to 'v8::Object::Set(v8::Persistent<v8::String>&, v8::Local<v8::Object>, v8::PropertyAttribute)'
/tmp/v8js/v8js.cc:640:93: note: candidates are:
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:2036:8: note: bool v8::Object::Set(v8::Handle<v8::Value>, v8::Handle<v8::Value>, v8::PropertyAttribute)
/usr/include/v8.h:2036:8: note:   no known conversion for argument 1 from 'v8::Persistent<v8::String>' to 'v8::Handle<v8::Value>'
/usr/include/v8.h:2040:8: note: bool v8::Object::Set(uint32_t, v8::Handle<v8::Value>)
/usr/include/v8.h:2040:8: note:   candidate expects 2 arguments, 3 provided
/tmp/v8js/v8js.cc: In function 'void php_v8js_timer_thread()':
/tmp/v8js/v8js.cc:706:14: warning: 'v8::Locker::Locker()' is deprecated (declared at /usr/include/v8.h:5173) [-Wdeprecated-declarations]
/tmp/v8js/v8js.cc: In function 'void zim_V8Js_executeString(int, zval*, zval**, zval*, int)':
/tmp/v8js/v8js.cc:754:2: error: no matching function for call to 'v8::Context::Scope::Scope(v8::Persistent<v8::Context>&)'
/tmp/v8js/v8js.cc:754:2: note: candidates are:
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:5046:5: note: v8::Context::Scope::Scope(v8::Isolate*, v8::Persistent<v8::Context>&)
/usr/include/v8.h:5046:5: note:   candidate expects 2 arguments, 1 provided
/usr/include/v8.h:5042:14: note: v8::Context::Scope::Scope(v8::Handle<v8::Context>)
/usr/include/v8.h:5042:14: note:   no known conversion for argument 1 from 'v8::Persistent<v8::Context>' to 'v8::Handle<v8::Context>'
/usr/include/v8.h:5040:9: note: constexpr v8::Context::Scope::Scope(const v8::Context::Scope&)
/usr/include/v8.h:5040:9: note:   no known conversion for argument 1 from 'v8::Persistent<v8::Context>' to 'const v8::Context::Scope&'
/tmp/v8js/v8js.cc: In function 'void php_v8js_persistent_zval_ctor(zval**)':
/tmp/v8js/v8js.cc:905:4: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
/tmp/v8js/v8js.cc: In function 'void php_v8js_persistent_zval_dtor(zval**)':
/tmp/v8js/v8js.cc:918:4: warning: deprecated conversion from string constant to 'char*' [-Wwrite-strings]
/tmp/v8js/v8js.cc: At global scope:
/tmp/v8js/v8js.cc:1054:1: warning: narrowing conversion of '-1' from 'int' to 'zend_uint {aka unsigned int}' inside { } [-Wnarrowing]
/tmp/v8js/v8js.cc:1068:1: warning: narrowing conversion of '-1' from 'int' to 'zend_uint {aka unsigned int}' inside { } [-Wnarrowing]
/tmp/v8js/v8js.cc:1071:1: warning: narrowing conversion of '-1' from 'int' to 'zend_uint {aka unsigned int}' inside { } [-Wnarrowing]
/tmp/v8js/v8js.cc: In function 'void php_v8js_write_property(zval*, zval*, zval*, const zend_literal*)':
/tmp/v8js/v8js.cc:1089:2: error: no matching function for call to 'v8::Context::Scope::Scope(v8::Persistent<v8::Context>&)'
/tmp/v8js/v8js.cc:1089:2: note: candidates are:
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:5046:5: note: v8::Context::Scope::Scope(v8::Isolate*, v8::Persistent<v8::Context>&)
/usr/include/v8.h:5046:5: note:   candidate expects 2 arguments, 1 provided
/usr/include/v8.h:5042:14: note: v8::Context::Scope::Scope(v8::Handle<v8::Context>)
/usr/include/v8.h:5042:14: note:   no known conversion for argument 1 from 'v8::Persistent<v8::Context>' to 'v8::Handle<v8::Context>'
/usr/include/v8.h:5040:9: note: constexpr v8::Context::Scope::Scope(const v8::Context::Scope&)
/usr/include/v8.h:5040:9: note:   no known conversion for argument 1 from 'v8::Persistent<v8::Context>' to 'const v8::Context::Scope&'
/tmp/v8js/v8js.cc:1094:63: error: no matching function for call to 'v8::Object::Get(v8::Persistent<v8::String>&)'
/tmp/v8js/v8js.cc:1094:63: note: candidates are:
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:2054:16: note: v8::Local<v8::Value> v8::Object::Get(v8::Handle<v8::Value>)
/usr/include/v8.h:2054:16: note:   no known conversion for argument 1 from 'v8::Persistent<v8::String>' to 'v8::Handle<v8::Value>'
/usr/include/v8.h:2056:16: note: v8::Local<v8::Value> v8::Object::Get(uint32_t)
/usr/include/v8.h:2056:16: note:   no known conversion for argument 1 from 'v8::Persistent<v8::String>' to 'uint32_t {aka unsigned int}'
/tmp/v8js/v8js.cc: In function 'void php_v8js_unset_property(zval*, zval*, const zend_literal*)':
/tmp/v8js/v8js.cc:1106:2: error: no matching function for call to 'v8::Context::Scope::Scope(v8::Persistent<v8::Context>&)'
/tmp/v8js/v8js.cc:1106:2: note: candidates are:
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:5046:5: note: v8::Context::Scope::Scope(v8::Isolate*, v8::Persistent<v8::Context>&)
/usr/include/v8.h:5046:5: note:   candidate expects 2 arguments, 1 provided
/usr/include/v8.h:5042:14: note: v8::Context::Scope::Scope(v8::Handle<v8::Context>)
/usr/include/v8.h:5042:14: note:   no known conversion for argument 1 from 'v8::Persistent<v8::Context>' to 'v8::Handle<v8::Context>'
/usr/include/v8.h:5040:9: note: constexpr v8::Context::Scope::Scope(const v8::Context::Scope&)
/usr/include/v8.h:5040:9: note:   no known conversion for argument 1 from 'v8::Persistent<v8::Context>' to 'const v8::Context::Scope&'
/tmp/v8js/v8js.cc:1111:63: error: no matching function for call to 'v8::Object::Get(v8::Persistent<v8::String>&)'
/tmp/v8js/v8js.cc:1111:63: note: candidates are:
In file included from /tmp/v8js/php_v8js_macros.h:30:0,
                 from /tmp/v8js/v8js.cc:28:
/usr/include/v8.h:2054:16: note: v8::Local<v8::Value> v8::Object::Get(v8::Handle<v8::Value>)
/usr/include/v8.h:2054:16: note:   no known conversion for argument 1 from 'v8::Persistent<v8::String>' to 'v8::Handle<v8::Value>'
/usr/include/v8.h:2056:16: note: v8::Local<v8::Value> v8::Object::Get(uint32_t)
/usr/include/v8.h:2056:16: note:   no known conversion for argument 1 from 'v8::Persistent<v8::String>' to 'uint32_t {aka unsigned int}'```
any clue? )
``````

--- a/README.md
+++ b/README.md
@@ -9,7 +9,7 @@ The extension allows you to execute Javascript code in a secure sandbox from PHP
 Minimum requirements
 --------------------
 
-- V8 Javascript Engine library (libv8) version 3.2.4 or above <http://code.google.com/p/v8/> (trunk)
+- V8 Javascript Engine library (libv8) master <https://github.com/v8/v8/> (trunk)
 
 	V8 is Google's open source Javascript engine.
 	V8 is written in C++ and is used in Google Chrome, the open source browser from Google.
@@ -20,6 +20,36 @@ Minimum requirements
 
   This embedded implementation of the V8 engine uses thread locking so it should work with ZTS enabled.
   However, this has not been tested yet.
+  
+COMPILING LATEST VERSION
+========================
+
+Compile latest v8
+-----------------
+
+```
+cd /tmp
+git clone https://github.com/v8/v8.git
+cd v8
+make dependencies
+make native library=shared -j8
+sudo cp out/native/lib.target/libv8.so /usr/lib/libv8.so
+sudo cp include/v8* /usr/include
+
+```
+
+Compile php-v8js itself
+-----------------------
+
+```
+cd /tmp
+git clone https://github.com/preillyme/v8js.git
+cd v8js
+phpize
+./configure CXXFLAGS="-DV8_USE_UNSAFE_HANDLES=1"
+make
+sudo make install
+```
 
 
 PHP API

